processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -144,15 +161,13 @@ public SearchQueryQueryToolChest(
         final byte[] querySpecBytes = query.getQuery().getCacheKey();
         final byte[] granularityBytes = query.getGranularity().cacheKey();
 
-        final Collection<DimensionSpec> dimensions = query.getDimensions() == null
-                                                     ? ImmutableList.<DimensionSpec>of()
-                                                     : query.getDimensions();
-
-        final byte[][] dimensionsBytes = new byte[dimensions.size()][];
+        final List<DimensionSpec> dimensionSpecs =
```

Comment:
I think if we reuse the dimensionSpecs from the enclosing CacheStrategy,
we should also reuse `query` from the enclosing CacheStrategy to avoid inconsistency among querySepcBytes, granularityBytes and dimensionSpecs calculation.
Then, parameter `query` of computeCacheKey() will be ignored. Is it OK?